lib/string_decoder.js

```diff
@@ -57,8 +60,16 @@ function StringDecoder(encoding) {
       this.end = base64End;
       nb = 3;
       break;
-    default:
-      this.write = simpleWrite;
+    case 'hex':
+      this.write = hexText;
+      this.end = simpleEnd;
+      return;
+    case 'latin1':
+      this.write = latin1Text;
+      this.end = simpleEnd;
+      return;
+    case 'ascii':
+      this.write = asciiText;
```

Comment:
These changes concern me a bit because they make the function size exceed Crankshaft's max inlineable source size.